assets/controllers/ajax_toggle_controller.js

```diff
@@ -4,28 +4,30 @@ import {Tooltip} from "bootstrap-next";
 
 export default class extends Controller {
   static values = {url: String};
-  static targets = ['icon', 'switch']
+  static targets = ['icon', 'switch', 'leftBar'];
 
   toggle(event) {
     const icons = this.iconTargets;
     const toggleSwitch = this.switchTarget;
+    const leftBar = this.leftBarTarget;
 
     axiosInstance.patch(this.urlValue)
       .then(function () {
         icons.forEach(
           (icon) => {
-              updateTooltipTitle(icon);
-              icon.dataset.toggleClasses.split(' ').forEach(
-                  (cssClass) => icon.classList.toggle(cssClass)
-              );
+            updateTooltipTitle(icon);
+            icon.dataset.toggleClasses.split(' ').forEach(
+              (cssClass) => icon.classList.toggle(cssClass)
+            );
           }
-        )
+        );
 
         updateTooltipTitle(toggleSwitch);
+        updateLeftBarColor(leftBar);
       })
```

Comment:
Maybe you could use data attributes in order to  change color (same way as icons classes are toggled).
```
//template
data-toggle-classes="shared not-shared"

//controller
leftBar.dataset.toggleClasses.split(' ').forEach((cssClass) => leftBar.classList.toggle(cssClass))
```